test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1732,8 +1736,17 @@ class test_ucp_am_nbx_rndv : public test_ucp_am_nbx_prereg {
         return cfg->rndv_frag_size[mem_type];
     }
 
+    void check_rma_support()
+    {
+        if (!sender().is_rndv_supported()) {
+            UCS_TEST_MESSAGE << "RNDV is not supported";
+            m_check_recv_rndv_flags = false;
```

Comment:
why? what if this is AM-based rndv?
also if rndv is not supported this test suite should be skipped

Reply:
Maybe message confused you. That case is created to handler AM-based RNDV case. AFAIR AM RNDV doesn't set `recv_attr` flags. 

I can change message here to "RMA is not supported" would it be better from your point of view?

Reply:
what flag do you mean? AM-based RNDV is using generic flow RTS->RTR->data chunks

Reply:
My bad, you are right. The thing here is that due to new logic if `RNDV_SCHEME=get_zcopy` and rndv/get/zcopy is unavailable eager can be selected instead of RNDV.

So we cannot rely on fact that `UCP_AM_RECV_ATTR_FLAG_RNDV` and `UCP_AM_RECV_ATTR_FLAG_DATA`  will be set in `am_data_handler`

Reply:
eager should not be selected because we set RNDV_THRESH in  `test_ucp_am_nbx_rndv`
Please either remove `check_rma_support` at all, or skip the test if RMA is not supported

Reply:
can we check sender().is_rndv_supported() directly in the callback and avoid m_check_recv_rndv_flags variable?

Reply:
Together with @brminich we've decided to skip this test if RNDV not supported. @yosefe does this option work for you? `m_check_recv_rndv_flags` still needed for `test_ucp_am_nbx_rndv_ppln.host_buff_host_frag` test